script.js

```diff
@@ -1,4 +1,275 @@
+var player1DiceRoll1 = 0;
+var player1DiceRoll2 = 0;
+var player2DiceRoll1 = 0;
+var player2DiceRoll2 = 0;
+var player1FinalDiceRollResult = 0;
+var player2FinalDiceRollResult = 0;
+var finalPlayer1AdvancedResults = 0;
+var finalPlayer2AdvancedResults = 0;
+var player1AdvanceRoll = [];
+var player2AdvanceRoll = [];
+var myOutputValue = "";
+var player1WinCount = 0;
+var player2WinCount = 0;
+var currentGameMode = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  //Output Gamemode choices "normal" and "advanced"
+  if (currentGameMode == "") {
+    myOutputValue =
+      "Please choose between the two gamemodes. Submit 'normal' for normal gamemode (i.e., 2 dices), and 'advanced' for advanced gamemode! (i.e., select no. of dices to be played, autochoice)";
+    currentGameMode = "Gamemode selected";
+    return myOutputValue;
+    //User selects Gamemode via gameMode function
+  } else if (currentGameMode == "Gamemode selected") {
+    var selectGameMode = gameMode(input);
+    myOutputValue = selectGameMode;
+    return myOutputValue;
+    //If gamemode is "Normal"
+  } else if (currentGameMode == "Submit") {
+    player1DiceRoll1 = rollDice();
+    player1DiceRoll2 = rollDice();
```

Comment:
You can consider putting this in a helper function to simplify the logic in your main function even more.